### PR TITLE
Reshuffle documentation

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* Shuffles documentation around into clearer IA
* Removes the sidebar config (that wasn't going to work really, it was getting too big)
* Adds intro page to the entire site giving clear roles on the IA spaces
* Adds intro page to the `docs` section to explain why it's set up the way it's setup (at the moment)

### Why is it needed?

* There have been _several_ complaints about how the contributor docs are confusing for people, how they're not sure "where things go" – i'm hoping this makes it somewhat clearer

### NOTE
* I wrote this really fast, so there may be spelling mistakes, please help by making suggestions if you spot them not just tell me 😅 
* It works better if you check the vercel preview to get a feel for everything

### CODE REVIEW

**Comment by Boegie19** [comment]
I think someting broke the auto preview generation nice.

**Comment by joshuaellis** [comment]
> I think someting broke the auto preview generation

it might have done, i'm not sure what "auto preview generation" is which is a bad sign 😅

**Comment by Boegie19** [comment]
I meant the preview vercel generates of every docs pr trough GitHub actions example is my docs pr
https://github.com/strapi/strapi/pull/16571

**Comment by Boegie19** [comment]
Thanks and now this makes match more sense.

File: docs/docs/docs/00-intro.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+title: Introduction
+---
+
+Hello & welcome to the documentation of the Strapi Monorepo! Here you'll find both technical and conceptual documentation on the codebase.
```
Comment (gu-stav) [inline docs/docs/docs/00-intro.md:5]:
```suggestion
Hello & welcome to the contributor documentation of the Strapi Monorepo! Here you'll find both technical and conceptual documentation on the codebase.
```

**Review by gu-stav — COMMENTED** [review]
One small suggestion - everything else works well for me. Thanks!

File: docs/docs/docs/00-intro.md (hunk 1 above)
Comment (joshuaellis) [reply docs/docs/docs/00-intro.md]: good point!

**Review by gu-stav — DISMISSED** [review]

File: docs/docs/index.md [hunk 2]
```diff
@@ -4,3 +4,23 @@ sidebar_label: Introduction
...
 
 # Strapi contributor documentation
+
+Welcome to the Strapi Contributor documentation.
+
+This documentation site is a constant WIP so please, continue to add and improve these docs. The general layout focusses on 4 key areas:
+
+## Guides
+
+This is where you'll probably want to start off. We have our contributing guides & code of conduct which are very important to read. There's also useful guides on common situations whilst developing such as ["Working with the Design System"](/guides/working-with-the-design-system). Or more high level guides such as best practices for frontend development (coming soon).
```
Comment (innerdvations) [inline docs/docs/index.md:14]:
```suggestion
This is where you'll probably want to start off. We have our contributing guides & code of conduct which are very important to read. There are also useful guides on common situations whilst developing such as ["Working with the Design System"](/guides/working-with-the-design-system) and higher-level guides such as best practices for frontend development (coming soon).
```

File: docs/docs/index.md [hunk 3]
```diff
@@ -4,3 +4,23 @@ sidebar_label: Introduction
...
+
+## Docs
+
+Within the docs section we have a multitude of both technical and conceptual documentation diving deep into particular parts of the Strapi monorepo that may not make as much sense just reading the code like ["Relations reordering in the database"](/docs/core/database/relations/reordering). There's also usage documentation for various pieces of code such as the [useDragAndDrop](/docs/core/content-manager/hooks/use-drag-and-drop) hook.
```
Comment (innerdvations) [inline docs/docs/index.md:18]:
```suggestion
Within the docs section we have a multitude of both technical and conceptual documentation diving deep into particular parts of the Strapi monorepo that may not make as much sense as just reading the code, like ["Relations reordering in the database"](/docs/core/database/relations/reordering). There's also usage documentation for various pieces of code such as the [useDragAndDrop](/docs/core/content-manager/hooks/use-drag-and-drop) hook.
```

File: docs/docs/index.md [hunk 4]
```diff
@@ -4,3 +4,23 @@ sidebar_label: Introduction
...
+
+## API Reference
+
+An advanced deep dive into some of the core driving classes of Strapi with explanations on the methods & parameters available on commonly exposed classes as well as examples to compliment them for easier understanding.
+
+## RFCs
+
+A growing section we intend to populate over time with public facing RFCs once approve as a record. These assist in understanding the design direction of features and code to understand the contextual "whys" that may not be apparent.
```
Comment (innerdvations) [inline docs/docs/index.md:26]:
```suggestion
A growing section we intend to populate over time with public-facing RFCs once approved to maintain as a record. These assist in understanding the design direction of features and code to understand the contextual "whys" that may not be apparent.
```

**Review by innerdvations — DISMISSED** [review]

**Review by Marc-Roig — APPROVED** [review]

**Comment by joshuaellis** [comment]
Thanks for the checks so far folks, i'll squash this once everyone has had a read through / tomorrow some time 👍🏼

**Review by nathan-pichon — APPROVED** [review]
Thanks for this PR Josh! A great step forward more concise and clear documentation!

PS: I think you can do a "Squash and merge" when merging the PR. I don't know if it was what you meant by "I will squash this once everyone has had a read".

**Comment by derrickmehaffy** [comment]
Nice job, this makes it much clearer. The intention is to have the docs structure like our packages right?

**Comment by joshuaellis** [comment]
Yeah the docs section of the site, I think each package can do what it thinks is necessary for now until it becomes unclear // confused and we can readdress it using the material as a guide.

**Pull request merged**